Add support for NiceError stack traces in syslog

VanWeapon opened the pull request:
Turn stack trace logs from `NiceError` into clickable links. Log messages produced by NiceError contain [table_name:sys_id] info for each call in the call stack and so can produce more helpful logs and errors.  Turning these into clickable links speeds up developer workflow
see #325 

I have tested by running the below function in the console while sitting on `syslog_list.do` and it seemed to work fine.

arnoudkooi commented:
Thanks but doesn't seem to be working as expected...

![Google Chrome20221205 - 1020](https://user-images.githubusercontent.com/840781/205600469-0d0da8d3-253d-4299-bae3-dd35db91286b.png)

VanWeapon commented:
Good point, 
I've updated the function to below which ensures 32 character sys_ids before it will match: 
```js
jQuery("td.vt:contains('-------- STACK TRACE ---------')").each(function (
            tableCellIndex,
            tableCell
        ) {
            var regex = /(([a-z_]+):([a-z0-9]{32}))/gm;
            var found = tableCell.innerText.match(regex);
            if (found !== null) {
                found.forEach(function (find) {
                    var str = find.split(":");
                    var t = str[0];
                    var id = str[1];
                    var newHtml = tableCell.innerHTML.replaceAll(
                        find,
                        `<a title="Link via SN Utils" href='/${t}.do?sys_id=${id}'>${find}</a>`
                    );
                    tableCell.innerHTML = DOMPurify.sanitize(newHtml, { ADD_ATTR: ["target"] });
                });
            }
        });
```

arnoudkooi commented:
I'll check later. I'm slowly getting rid of jQuery functions, so don't want to introduce new usage.
And I like the arrow function inside de forEach loop ie. `forEach(find => {})`

arnoudkooi commented:
Tested your code and does not seem to convert any link now...

VanWeapon commented:
My addition is intended to work with [table_name:sys_id] syntax, that's generated for you with the `NiceError` script include. 
usage: 
```js
var msg = new global.NiceError(message);		
gs.info(msg);
```
When i paste that jquery function into the console i get the below results:

![image](https://user-images.githubusercontent.com/30616047/205748203-c9781811-3b45-424f-b536-aae07c83949a.png)


Maybe we could generically add support for both `table_name.sys_id.field` and `[table_name:sys_id]` patterns and not worry about checking for certain text in the cell? 

I only used jquery since that pattern was already in use for the other regex being done, if it can be converted to native that would be even better, probably would run faster too

VanWeapon commented:
Oh that's interesting, the `.raise()` method produces a different looking stack trace compared to just generating the log with the constructor function. 

My log was produced using this pattern:
```js
var msg = new NiceError('some log message');
gs.info(msg);
```

VanWeapon commented:
Also running via a background script wont have a stack trace because it's not passing through any other records in the system.

When i call the script include from the above screenshots in a background script, i get the below logs:

![image](https://user-images.githubusercontent.com/30616047/206137870-7a344532-9886-4050-86eb-526668657209.png)

arnoudkooi commented:
I saw .raise() in GlideQuery while doing a code search.
Not using it myself.
Will you do a new PR, think both ways should be supported to be consistent.
